sgxs-tools/src/sgx_detect/main.rs

```diff
@@ -248,6 +265,52 @@ impl SgxSupport {
             }
             Ok(Rc::new(RefCell::new(lib.build())))
         })();
+        let gsgxdev= match File::open("/dev/gsgx") {
+            Ok(_) => Ok(()),
+            Err(e) => Err(e.into()),
+        };
+        let nodeagent_status = (|| {
+            let request_url = format!("http://localhost:9092/v1/sys/version");
+            let mut response = reqwest::get(&request_url)?;
+            let ver :Version = response.json().unwrap();
+            Ok(NodeAgentStatus{current_ver: ver.version.clone()})
+        })();
+        let permdaemon_status = (|| -> Result<(), Error> {
+            let mut isgx = false;
+            let mut gsgx = false;
+            let mut installed = false;
+            let mut system = sysinfo::System::new();
+            system.refresh_all();
+            for proc in system.get_process_by_name("bash") {
+                if (proc.cmd().len() >= 2) && (installed == false) {
+                    installed = proc.cmd()[1].contains("sgx_perm_daemon");
+                    if installed {
+                        break;
+                    }
+                }
+            }
+            let stdout = Command::new("journalctl")
+                .stdout(Stdio::piped())
+                .spawn()?
+                .stdout
+                .ok_or_else(|| IOError::new(ErrorKind::Other,"[perm_daemon_test]Could not capture the journalctl output."))?;
+            BufReader::new(stdout)
+                .lines()
+                .filter_map(|line| line.ok())
+                .filter(|line| line.find("sgx_perm_daemon").is_some())
+                .for_each(|line| {
+                if line.contains("gsgx: 10:54") {
+                    gsgx = true;
+                } else if line.contains("isgx: 10:55"){
+                    isgx = true;
+                }
+            });
+            if isgx && gsgx && installed {
```

Comment:
You shouldn't be checking the dependencies here. There's the dependency mechanism for that.

Reply:
this is different from the dependency check in the `tests/mod.rs` where there is a separate test for each dependency. In this case the 3 conditions for `perm_daemon` are being checked. It makes little sense to write 3 separate tests just to check this.

Reply:
It's not different. perm_daemon is just another component of the whole system. You already have a `PermDaemon` test in the tree, so it's easy to add isgx and gsgx as a dependency there.

Reply:
I now see this relates to the checking of the logs, not the loading of the drivers. Please ignore my previous comment.